### ISSUE TITLE
Doesn't work on Ubuntu 22.

### ISSUE THREAD

**Issue by lprkhb** [philipturner/swift-colab#25]
Doesn't work properly on Ubuntu 22(Google Colab) :/.

I tested with 5.6.2 and 5.7.3 Swift versions.

P.S. Anyway Thx! for your project.

**Comment by philipturner** [comment]
Yeah, it’s been broken for over a year. Didn’t have much motivation to fix it. Probably requires a lot of work.

**Comment by philipturner** [comment]
I think practically, for cross-platform Swift, the best method is VS Code + Swift extension. GitHub gists can serve a similar role to Colab notebooks. Just learn how to write documentation and tutorials in source code comments. With the constraint that every line is <80 characters.

Alternatively, use Markdown with simple pieces of code that the user copies into their IDE. Thoroughly test that your code works, minimize the number of dependencies, and explain how to download each dependency. Example: https://github.com/philipturner/HDL/blob/main/Documentation/GrapheneSiliceneBilayer.md

Another document structured similarly to Jupyter notebooks. This proves that PythonKit still has immense utility: https://github.com/philipturner/molecular-renderer/blob/main/Sources/HardwareCatalog/Simulation/DMSTooltips/README.md

**Comment by lprkhb** [comment]
I'll try to fix the problem. Although i don't know how much we need to get deeper into it. It's a shame the backward-compatibility is broken by Google.

**Comment by philipturner** [comment]
It's a lot of debugging through custom I/O ports. It was very time-consuming. You have to ensure the entire program initializes in under 30 seconds, to get fast feedback loops while debugging.

**Comment by lprkhb** [comment]
Probably, i have to write a message to Google first ;-)

**Comment by philipturner** [comment]
Swift isn't a priority use case for them. The current state of Swift-Colab isn't really their fault or their obligation to fix.

**Comment by lprkhb** [comment]
Ok. P.S. Btw cool stuff on your desk. I wonder to become obsessed with Metal and M1 Chip Graphics one day.

**Comment by philipturner** [comment]
It’s more than just that now. Evolved into expertise in high-performance computing in general. Soon to have access to a 1 petaFLOPS supercluster to use for non-ML workloads :)

**Comment by lprkhb** [comment]
What can be more beautiful than ML and DeepLearning tasks? Perhaps only Bio and Nano systems :D

**Comment by philipturner** [comment]
Solving one of the greatest computational challenges of computational chemistry: accurate and efficient an initio simulations. The solution is so beautiful.
